packages/ember-metal/lib/mixin.js

```diff
@@ -257,8 +257,12 @@ function mergeMixins(mixins, m, descs, values, base, keys) {
 
 let IS_BINDING = /^.+Binding$/;
 
-function detectBinding(obj, key, value, m) {
-  if (IS_BINDING.test(key)) {
+export function detectBinding(obj, key, value, m) {
+  if (key.length < 8) { return false; }
+
+  let lastIndex = key.indexOf('Binding', key.length - 7);
```

Comment:
ah if anyone cares to explore further (i was nerd sniped) 

see: https://github.com/stefanpenner/perf-scratch/commit/a57d6f9dcd9332928b813951c5a5774f3af0fe05